src/openforms/registrations/contrib/zgw_apis/plugin.py

```diff
@@ -239,8 +239,21 @@ def register_submission(
         )
 
         for attachment in submission.attachments:
+            # attribute names conform to the Open Documenten API specification
             iot = attachment.informatieobjecttype or options["informatieobjecttype"]
-            doc_options = {**options, "informatieobjecttype": iot}
+            bron = attachment.bronorganisatie or options["organisatie_rsin"]
+            vetrouwelik = (
+                attachment.vertrouwelijkheidaanduiding
+                or options["vertrouwelijkheidaanduiding"]
+            )
+            titel = attachment.titel or options.get("titel")
```

Comment:
`None` is not an accepted type for the API spec, and furthermore a non-empty string is required. So let's keep at least the type consistent:

```suggestion
            titel = attachment.titel or options.get("titel", "")
```